packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart

```diff
@@ -933,6 +935,32 @@ if %errorlevel% neq 0 goto :VCEnd</Command>
     FileSystemUtils: () => FileSystemUtils(fileSystem: fileSystem, platform: windowsPlatform),
     Usage: () => usage,
   });
+
+  // Confirms that running for Windows in a directory with a
+  // bad caracter (' in this case) throws the desired error message
```

Comment:
If whoever is responsible (CMake or Visual Studio) ever fixes this bug, we should remove the check & this test. I would consider adding a link to [the issue](https://github.com/flutter/flutter/issues/104802) in this comment for our own future reference.